deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/metadata/ManifestGenerator.kt

```diff
@@ -0,0 +1,87 @@
+package com.airbnb.deeplinkdispatch.metadata
+
+import androidx.room.compiler.processing.XFiler
+import androidx.room.compiler.processing.XMessager
+import androidx.room.compiler.processing.XProcessingEnv
+import com.airbnb.deeplinkdispatch.DeepLinkAnnotatedElement
+import com.airbnb.deeplinkdispatch.base.ManifestGeneration
+import com.airbnb.deeplinkdispatch.metadata.writers.ManifestWriter
+import com.airbnb.deeplinkdispatch.metadata.writers.Writer
+import java.io.PrintWriter
+import java.io.StringWriter
+import java.nio.file.Path
+import javax.tools.Diagnostic
+
+/**
+ * This generates the input manifest for the manifest merger in the Gradle plugin.
+ *
+ * It creates a manifest file that contains all the intent filters from the annotated elements.
+ * The manifest is written to KSP's resource output directory using the filer API, which ensures
+ * proper caching by Gradle.
+ *
+ * The Gradle plugin reads from: build/generated/ksp/<variant>/resources/deeplinkdispatch/AndroidManifest.xml
+ */
+internal class ManifestGenerator(
+    private val processingEnv: XProcessingEnv,
+) {
+    private val messager: XMessager = processingEnv.messager
+
+    // Track if we've already written the manifest to avoid duplicate writes in multi-round processing
+    private var manifestWritten = false
+
+    fun write(elements: List<DeepLinkAnnotatedElement>) {
+        // Skip if we've already written the manifest in a previous round
+        if (manifestWritten) {
```

Comment:
I _think_ this is not the idiomatic way with ksp. With the nature of rounds, some new deeplink elements could be picked up in later rounds (ie, if someone generates them with another processor), and with this approach they would be missed.

I am a bit rusty on this, but I think with an aggregating processor like this it would be preferable to collect all of the data by appending to it each round and then writing the final file in the on finish callback (or just document that we don't support multi round)